File: apk/app/src/main/java/com/example/momentra/data/repository/TransactionRef.kt
package com.example.momentra.data.repository

import com.example.momentra.data.api.ActivityItemDto

enum class TransactionDomain {
    PERSONAL,
    GROUP,
    BUSINESS,
}

enum class TransactionResourceType {
    EXPENSE,
    INCOME,
    CONTRIBUTION,
    REVENUE,
}

/** Unified client handle for governed transaction CRUD (QH-T1). */
data class TransactionRef(
    val domain: TransactionDomain,
    val resourceType: TransactionResourceType,
    val resourceId: String,
    val momentId: String,
) {
    companion object {
        fun fromActivity(momentId: String, item: ActivityItemDto): TransactionRef? {
            val payload = item.activityPayload ?: return null
            payload.expenseId?.let {
                return TransactionRef(TransactionDomain.PERSONAL, TransactionResourceType.EXPENSE, it, momentId)
            }
            payload.incomeId?.let {
                return TransactionRef(TransactionDomain.PERSONAL, TransactionResourceType.INCOME, it, momentId)
            }
            return null
        }
    }
}

File: momentra/momentra/Data/TransactionRef.swift
import Foundation

enum TransactionDomain: String {
    case personal = "PERSONAL"
    case group = "GROUP"
    case business = "BUSINESS"
}

enum TransactionResourceType: String {
    case expense = "EXPENSE"
    case income = "INCOME"
    case contribution = "CONTRIBUTION"
    case revenue = "REVENUE"
}

/// Unified client handle for governed transaction CRUD (QH-T1).
struct TransactionRef: Equatable {
    let domain: TransactionDomain
    let resourceType: TransactionResourceType
    let resourceId: String
    let momentId: String

    static func fromActivity(momentId: String, item: APIClient.ActivityItemPayload) -> TransactionRef? {
        if let expenseId = item.activityPayload?.expenseId {
            return TransactionRef(domain: .personal, resourceType: .expense, resourceId: expenseId, momentId: momentId)
        }
        if let incomeId = item.activityPayload?.incomeId {
            return TransactionRef(domain: .personal, resourceType: .income, resourceId: incomeId, momentId: momentId)
        }
        return nil
    }
}
